Assets/Scripts/Model/Weapon.cs

```diff
@@ -63,33 +63,48 @@ public float getCurrentDamage()
 	// Add target to list
 	public void addTarget(CanReceiveDamage target){
 		this.targets.Add (target);
-		Debug.Log ("Targets to attack :" + targets.Count);
+		Debug.Log (this.gameObject.name + "-> Targets to attack :" + targets.Count);
 	}
 
 	// Remove target from list
 	public void removeTarget(CanReceiveDamage target){
 		this.targets.Remove (target);
+		Debug.Log (this.gameObject.name + "-> Targets to attack :" + targets.Count);
 	}
 
-	// Called to attack a target
-	public void Attack()
-	{
+	// Get the available target to attack from the targets list
+	public CanReceiveDamage getAvailableTarget(){
+
 		// Checks if there is a target in the range
-		if (this.targets.Count > 0) {
+		while (this.targets.Count > 0) {
 
 			// Get target to attack
-			CanReceiveDamage target = targets [0];
+			CanReceiveDamage target = this.targets [0];
+
+			// Check if target is already dead
+			if (target.getCurrentHealth() <= 0.0f) {
```

Comment:
@kkaren Don't you have the same problem as before? That if the Object is Destroyed, than it's null so you can't access to its attribute. Why don't you just check (target == null) and that's it?